tests/v2/validation/crdgeneration/crd_generation.go

```diff
@@ -0,0 +1,42 @@
+package crdgeneration
```

Comment:
yep agreed but the only thing I want to change here is instead of crd package we can change this to publicAPI instead and the generation_test.go which will be helpful for the publicAPI automation.